File: deal/src/main/java/ru/artemev/deal/service/DealService.java
package ru.artemev.deal.service;

import ru.artemev.deal.dto.FinishRegistrationRequestDTO;
import ru.artemev.deal.dto.LoanApplicationRequestDTO;
import ru.artemev.deal.dto.LoanOfferDTO;
import ru.artemev.deal.model.enums.ApplicationStatus;

import java.util.List;

public interface DealService {
  List<LoanOfferDTO> calculationPossibleLoans(LoanApplicationRequestDTO loanApplicationRequestDTO);

  void selectOneOfOffers(LoanOfferDTO loanOfferDTO);

  void completionOfRegistration(Long id, FinishRegistrationRequestDTO finishRegistrationRequestDTO);

  void sendDocuments(Long applicationId);

  void signDocuments(Long applicationId);

  void codeDocuments(Long applicationId, Integer sesCode);

  void updateApplicationStatus(Long applicationId, ApplicationStatus applicationStatus);
}

File: deal/src/main/kotlin/ru/kit/deal/service/DealService.kt
package ru.kit.deal.service

import ru.kit.deal.dto.FinishRegistrationRequestDTO
import ru.kit.deal.dto.LoanApplicationRequestDTO
import ru.kit.deal.dto.LoanOfferDTO

interface DealService {
    fun calculationPossibleLoans(loanApplicationRequestDTO: LoanApplicationRequestDTO): List<LoanOfferDTO>
    fun selectOneOfOffers(loanOfferDTO: LoanOfferDTO): Unit
    fun completionOfRegistration(id: Long, finishRegistrationRequestDTO: FinishRegistrationRequestDTO)
}